test/snippet/argument_parser/custom_argument_parsing_enumeration.cpp

```diff
@@ -0,0 +1,46 @@
+#include <system_error>
+
+#include <seqan3/argument_parser/all.hpp>
+
+namespace seqan3::custom
+{
+// Specialise the seqan3::custom::argument_parsing data structure to enable parsing of std::errc.
+template <>
+struct argument_parsing<std::errc>
+{
+    // Specialise a mapping from an identifying string to the respective value of your type Foo.
+    static inline const std::unordered_map<std::string_view, std::errc> enumeration_names
```

Comment:
```suggestion
    static inline std::unordered_map<std::string_view, std::errc> const enumeration_names
```

I think I commented this already :thinking: